keps/sig-api-machinery/NNNN-watch-list/README.md

```diff
@@ -0,0 +1,956 @@
+The mode is not used in informers today but is supported by every kube-apiserver for legacy, compatibility reasons.
+
+Note 2: informers need consistent lists to avoid time-travel when switching to another HA instance of kube-apiserver with outdated/lagging watch cache.
+See the following [issue](https://github.com/kubernetes/kubernetes/issues/59848) for more details.
+
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+The following sequence diagram depicts steps that are needed to complete the proposed feature.
+A high-level overview of each was provided in a table that follows immediately the diagram.
+Whereas further down in this section we provided a detailed description of each required step.
+
+![flow between an informer and the watch cache](./sequence_diagram.png "flow between an informer and the watch cache")
+<table>
+<tr>
+  <th>Step</th>
+  <th>Description</th>
+</tr>
+<tr>
+  <th>1.</th>
+  <th>The reflector establishes a WATCH request with the watch cache.</th>
+ </tr>
+<tr>
+  <th>2.</th>
+  <th>The watch cache contacts etcd for the most up-to-date ResourceVersion.</th>
+</tr>
+<tr>
+  <th>2a.</th>
+  <th>The watch cache starts streaming initial data. The data it already has in memory.</th>
+</tr>
+<tr>
+  <th>2b.</th>
+  <th>The watch cache waits until it has observed data up to the RV received in step 2. Streaming all new data (if any) to the reflector immediately.</th>
+</tr>
+<tr>
+  <th>2c.</th>
+  <th>The watch cache has observed the RV (from step 2) and sends a bookmark event with the given RV to the reflector.</th>
+</tr>
+<tr>
+  <th>3.</th>
+  <th>The reflector replaces its internal store with collected items, updates its internal resourceVersion to the one obtained from the bookmark event.</th>
+</tr>
+<tr>
+  <th>3a.</th>
+  <th>The reflector uses the WATCH request from step 1 for further progress notifications.</th>
+</tr>
+</table>
+
+Step 1: The reflector gets a snapshot of data from the server by passing RV=”” (= from watch cache) and setting resourceVersionMatch=ConsistentInitialAdds (= ensure freshness).
```

Comment:
Related to one of my commments above - I don't think it has to be RV="" - I think we can allow others and differentiate based on resourceVersionMatch param.

Reply:
clarified